pkg/kubectl/cmd/create_role.go

```diff
@@ -105,13 +116,23 @@ func (c *CreateRoleOptions) Complete(cmd *cobra.Command, args []string) error {
 	// e.g. --resource=pods,deployments.extensions
 	resources := cmdutil.GetFlagStringSlice(cmd, "resource")
 	for _, r := range resources {
```

Comment:
logically, this should split first on the `/` to get a subresource (if any), then split the first segment on `.` to get the resource and group